### DISCUSSION TITLE
Documentation Updates

### DISCUSSION THREAD

**Discussion by tuxxxed** [herbstluftwm/herbstluftwm#1573] [General]
Hi Thorsten,

I am now using HLWM as my daily wm.

HLWM is my third tiling wm with prior experience using Awesome and i3.

It is my favorite by far for the following reasons:
- simple concept and user interface - I feel like I am in control and not being controlled
- few dependencies and in particular no dependency on dBus and related systemd cruft
- configuration extensibility via any scripting language or even a compiled language
- elegant decorations and theming - simple frames and windows with border colors and line thicknesses
- a rich API for customizing layout, window management and workflow
- independent of status bar and run menu options - let the user decide those details
- vim style key bindings

All-in-all, I am very impressed with what has been developed and shared with the 'nix communities.

I still have much to learn and experiment with as my HLWM skills improve.  As part of that learning curve, I wonder if you are interested in documentation assistance?  I have found myself head scratching over the documentation on more than one occasion.  Though, after much introspection I have grokked many, not all, of the underlying concepts and ideas.

As a native English speaker and non-programmer, I think I might be able to simplify and clarify many of the underlying concepts, ideas and configuration details regarding HLWM.  And I suspect additional documentation would probably help others while indirectly increasing the attractiveness of HLWM for potential new users.  In terns of scale and scope, I am ultimately thinking of documentation on par with, if not better than, that provided by i3 and qtile.

From other posts, I know you are very busy person.  However, if you would be interested in collaborating on improved documentation please let me know.

All the best,
tuxxxed

**Comment by t-wissmann** [comment]
Hi tuxxxed,
nice, thank you, it would be really helpful to have better introductions especially for non-programming people. The man page is more in the style of an API-documentation. In my opinion, this is fine for man pages, but we should have something better for the website.
* There is [the tutorial](https://herbstluftwm.org/tutorial.html), but it covers only the absolute basics. Maybe this is a good place to start?
* Alternatively, you can say *where* you had expected to find a gentle introduction; maybe that's the best place to start :)
* Also, what I would like to have on the website is a kind of 'feature gallery' that lists features of hlwm, together with small screenshots / gifs (e.g. using `peek`) for demonstration, like this one (from PR #1381): ![hlwm-tabs](https://user-images.githubusercontent.com/9048813/142879398-82e3b117-9a61-406d-8f05-ab129b76eaa8.gif)

**Reply by tuxxxed** [reply]
Hi Thorsten,

It was very good to receive your positive reply and initial thoughts.

I concur with your thinking and see many paths forward.  Starting with the public facing website is a great first step and will benefit the largest number of users quickly.  Along the lines of best practices for technical documentation, it may be worth considering the work of the Diataxis.

Diataxis is a well formulated description of the different types of technical documentation and how each type is consumed by users and best generated by technical authors.  Their information can be accessed as follows on the web.

https://diataxis.fr/
https://github.com/evildmp/diataxis-documentation-framework

An independent article on the Diataxis approach can be found at the link below.
https://idratherbewriting.com/blog/what-is-diataxis-documentation-framework

From a practical perspective, what is the best way to proceed?  In particular, how would you prefer to continue the discussion, either publicly via github or another preferred method to facilitate dialog, idea exchange and draft work products?  I am open to most every form of online communication presently available.  This to include streaming video which may be very useful for initial introductions.  Just let me know how you would like to proceed.  In the interim, I will review the website to capture my thoughts.

I look forward to assisting you in these efforts and thank you for the opportunity to contribute.

All the best,

tuxxxed

**Reply by t-wissmann** [reply]
Well I'm not entirely sure whether this diataxis is now just 'an approach' or a concrete piece of software... In general, the best way for you to get started is by doing small changes regarding things which are not good at the moment. E.g. by changing the website or reorganizing them and then you can send a pull request. Such a pull request also includes a comment in which you can explain why you think the proposed change makes sense. The website is just a part of the main git repository.

**Reply by tuxxxed** [reply]
Hi Thorsten,

Apologies for the confusion.  Diataxis is an "Approach" to writing effective documentation.  It is not "Software".  The Diataxis approach is aimed specifically at improving "Technical Documentation".  The founders of the practice come form the tech industries and they sought to codify the approach by making it accessible to software developers and maintainers.  The github account is apparently their repository of web site content as I infer is also the case for the HLWM web site. 

More than anything else, Diataxis makes a clear distinction between four main categories of technical documentation and how they are each used by consumers and how they are best created and presented.  Those categories are:
- Tutorials
- How To Guides
- Explanations
- Reference

So for example, Man Pages are clearly "References" with detailed and dense information.  "Tutorials" are hands-on learning engagements.  "How-To Guides" are goal specific tasks such as setting up "Scratchpads" in HLWM.  And finally, "Explanations" are background information that seeks to aid understanding of what and why.  Each type of documentation is best approached in a specific manner, presentation and writing style so as not confuse the reader.  This is what the Diataxis website seeks to convey and distill to those writing documentation.  In other words a set of "Best Practices".

With respect to using github pull requests for website updates, I will need to learn how this process works to be effective.  I only just created an account as a placeholder and used it as a mechanism to reach out to you.  Please allow me a little time to learn what github can do.

If you have further suggestions, please let me know.

tuxxxed

**Reply by t-wissmann** [reply]
If the approach helps you to find the issues and gaps in the current website / documentation, then that's perfect. I understand that for non-programmers, it's a steep learning curve to use 'git' (and 'github'). But given that you're using herbstluftwm, I'm optimistic that you can master those.

Also, for getting started, I recommend watching the talk by Linux Torvalds on git (presented at google 10 years ago). I think he gets some ideas across, the talk is entirely non-technical, and it's absolutely entertaining to watch :-).

If you're not familiar with the difference between github and git: github.com is for 'git repositories' what youtube.com is for 'videos'.

**Reply by tuxxxed** [reply]
Hi Thorsten,

Thank you for the feedback and words of encouragement.  I have already started studying Git and Github documentation.  And I will add Torvalds talk to that list.  Learning about Github has always been on the technical to do list.  So, assisting with HLWM documentation is facilitating motivation for progress.  It will just take time to reach a base level of familiarity and competence.  Please be patient as I step through the learning process.

All the best,
Tuxxxed
